### ISSUE TITLE
pre-exec-cmd sometimes doesn't work / no output shown

### ISSUE THREAD

**Issue by erikbor** [riboseinc/rvc#213]
Hi @jjr840430 

I am troubleshooting some RVC problems we're experiencing and one of the issues is that our MFA (for use in `pre-exec-cmd`) scripts not work. I extracted one of the commands from the scripts to see if `rvd` would be able to execute it directly.

The command is `security find-generic-password -a test -s test-vpn -w`:

```
$ rvc test connect
name: test
	config-status: JSON configuration exists
	profile: /opt/rvc/etc/vpn.d/test.ovpn
	status: CONNECTING
	openvpn-status: DISCONNECTED
	in-total: 0
	out-total: 0
	timestamp: 1511774249
	auto-connect: Disabled
	pre-exec-cmd: security find-generic-password -a test -s test-vpn -w
	pre-exec-status: 
	pre-exec-interval: none

```

`rvd.log`:
```
2017-11-27T17:17:29HKT  INFO : CMD: Received command '{ "cmd": 0, "json": false, "param": "test" }'(at cmd.c:362)
2017-11-27T17:17:29HKT  INFO : CMD: Processing 'connect' command(at cmd.c:173)
2017-11-27T17:17:29HKT  INFO : VPN: Try to connect a VPN with name 'test'(at vpn.c:885)
2017-11-27T17:17:29HKT  INFO : VPN: Getting connection status for VPN connection with name 'test'(at vpn.c:1301)
2017-11-27T17:17:29HKT  INFO : VPN: Starting VPN connection with name 'test'(at vpn.c:797)
2017-11-27T17:17:29HKT  INFO : VPN: Getting status of VPN connection with name 'test'(at vpn.c:1071)
2017-11-27T17:17:29HKT  INFO : VPN: Running OpenVPN process for connection 'test'(at vpn.c:620)
2017-11-27T17:17:29HKT  INFO : CMD: Sending response(at cmd.c:126)
2017-11-27T17:17:29HKT  INFO : CMD: Response string is 
'name: test
        config-status: JSON configuration exists
        profile: /opt/rvc/etc/vpn.d/test.ovpn
        status: CONNECTING
        openvpn-status: DISCONNECTED
        in-total: 0
        out-total: 0
        timestamp: 1511774249
        auto-connect: Disabled
        pre-exec-cmd: security find-generic-password -a test -s test -w
        pre-exec-status: 
        pre-exec-interval: none
'
(code: 0)(at cmd.c:157)
2017-11-27T17:17:29HKT  INFO : VPN: Running pre-connect command 'security find-generic-password -a test -s test -w' with uid '501'(at vpn.c:455)
2017-11-27T17:17:29HKT  INFO : VPN: The status of pre-exec-cmd is '1'(at vpn.c:555)
```

As you can see there is no output logged while in case of an error the `security` utility on macOS would return something like this:
`security: SecKeychainSearchCopyNext: The specified item could not be found in the keychain.`
Or in case of a successful status a password would be shown.

`rvc status` will not even show the `1` in the `pre-exec-cmd`:
```
$ rvc status
name: test
	config-status: JSON configuration exists
	profile: /opt/rvc/etc/vpn.d/test.ovpn
	status: CONNECTING
	openvpn-status: DISCONNECTED
	in-total: 0
	out-total: 0
	timestamp: 1511774706
	auto-connect: Disabled
	pre-exec-cmd: security find-generic-password -a test -s test -w
	pre-exec-status: 
	pre-exec-interval: none

```

This is the `.json` file that belongs to the VPN:
```
$ sudo cat test.json 
Password:
{
  "name": "test",
  "auto-connect": false,
  "pre-connect-exec": "security find-generic-password -a test -s test -w"
}
```

**Comment by jjr840430** [comment]
Fixed by #215

**Comment by ronaldtse** [comment]
It works correctly now
